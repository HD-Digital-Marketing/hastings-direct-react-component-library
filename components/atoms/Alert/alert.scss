@import '../../../src/scss/variables/variables';

.a-alert {
	padding: 1em;
	margin-top: 1em;
	border-radius: 5px;
	width: 100%;
	box-sizing: border-box;
	background-color: $color-light;
	box-shadow: 0px 2px 3px rgba(0,0,0,0.1);
	border: 1px solid darken($color-light, 5%);

	&__icon {
		display: inline-block;
		width: 25px;
		height: 25px;
		background-color: #fff;
		border-radius: 50%;
		margin-right: 1em;
		line-height: 25px;
		text-align: center;
	}

	&--success {
		background-color: $color-success;
		color: #fff;
		border: 1px solid darken($color-success, 5%);

		.a-alert__icon {
			color: $color-success;
		}
	}

	&--info {
		background-color: $color-info;
		color: #fff;
		border: 1px solid darken($color-info, 5%);

		.a-alert__icon {
			color: $color-info;
		}
	}

	&--warning {
		background-color: $color-warning;
		color: #fff;
		border: 1px solid darken($color-warning, 5%);

		.a-alert__icon {
			color: $color-warning;
		}
	}

	&--error {
		background-color: $color-error;
		color: #fff;
		border: 1px solid darken($color-error, 5%);

		.a-alert__icon {
			color: $color-error;
		}
	}
}