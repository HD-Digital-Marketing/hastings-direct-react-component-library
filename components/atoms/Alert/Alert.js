import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './alert.scss'

const Alert = props => {

    const {
        theme = '',
        className = '',
        children
    } = props

    let alertIcon = '';

    const alertClass = classNames('a-alert', className, {
        'a-alert--success': theme === 'success',
        'a-alert--info': theme === 'info',
        'a-alert--warning': theme === 'warning',
        'a-alert--error': theme === 'error'
    })

    switch (theme) {
        case 'success':
            alertIcon = <span className="a-alert__icon" >&checkmark;</span>
            break;
        case 'warning':
            alertIcon = <span className="a-alert__icon">&#9888;</span>
            break;
        case 'info':
            alertIcon = <span className="a-alert__icon">&#8505;</span>
            break;
        case 'error':
            alertIcon = <span className="a-alert__icon">&#10005;</span>
            break;
        default:
            alertIcon = ''
    }

    return(
        <div className={alertClass}>
            {alertIcon}{children}
        </div>
    )
}

Alert.propTypes = {
    theme: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.string
}

export default Alert
