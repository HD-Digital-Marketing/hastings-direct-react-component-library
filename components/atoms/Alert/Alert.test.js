import React from 'react';
import renderer from 'react-test-renderer';
import Alert from './Alert';

describe('Check the alert renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(<Alert theme="error">This is an error message</Alert>)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });
});