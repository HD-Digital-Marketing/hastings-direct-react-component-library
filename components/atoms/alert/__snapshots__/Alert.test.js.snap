// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the alert renders it should match the snapshot 1`] = `
<div
  className="a-alert a-alert--error"
>
  <span
    className="a-alert__icon"
  >
    âœ•
  </span>
  This is an error message
</div>
`;
